src/main/java/core/service/handlers/BalanceHandler.java

```diff
@@ -0,0 +1,12 @@
+package core.service.handlers;
+
+import core.dao.FruitDao;
+import core.dao.FruitDaoImpl;
+
+public class BalanceHandler implements FruitOperationHandler {
+    @Override
+    public void doOperation(String fruitName, int fruitsQuantity) {
+        FruitDao fruitDao = new FruitDaoImpl();
```

Comment:
you should create a FrutiService class that will call methods of FruitDao, in handlers you should use FruitService, and you should initialize this varialbe in the constructor not inside of the method